examples/AzureSDKDemoSwift/Source/BlobDownloadViewController.swift

```diff
@@ -0,0 +1,277 @@
+        if indexPath.row == data.count - 1, noMoreData == false {
+            loadMoreSettings()
+        }
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath) as? CustomTableViewCell else { return }
+        guard let blobName = cell.keyLabel.text else { return }
+        guard let containerName = AppConstants.videoContainer else { return }
+        guard let blobClient = blobClient else { return }
+        guard let destination = try? blobClient.downloadUrl(
+            forBlob: blobName,
+            inContainer: containerName,
+            withOptions: AppState.downloadOptions
+        ) else {
+            showAlert(error: AzureError.general("Unable to create destination URL."))
+            return
+        }
+
+        let manager = FileManager.default
+        if let existingTransfer = downloadMap[indexPath] as? BlobTransfer {
+            // if transfer exists and is complete, open file, otherwise ignore
+            if manager.fileExists(atPath: destination.path), existingTransfer.incompleteBlocks == 0 {
+                playVideo(indexPath, destination)
+                return
+            }
+            return
+        } else {
+            // if no transfer exists but a file exists, play
+            if manager.fileExists(atPath: destination.path) {
+                playVideo(indexPath, destination)
+                return
+            }
+        }
+
+        // Otherwise, start the download with TransferManager.
+        let options = AppState.downloadOptions
+        do {
+            if let transfer = try blobClient.download(
+                blob: blobName,
+                fromContainer: containerName,
+                withRestorationId: "download",
+                withOptions: options
+            ) as? BlobTransfer {
+                downloadMap[indexPath] = transfer
+            }
+        } catch {
+            showAlert(error: error)
+        }
+        DispatchQueue.main.async { [weak self] in
+            self?.tableView.deselectRow(at: indexPath, animated: true)
+        }
+    }
+}
+
+extension BlobDownloadViewController: TransferDelegate {
+    func client(forRestorationId restorationId: String) -> PipelineClient? {
+        guard restorationId == "download" else { return nil }
+        return blobClient
+    }
+
+    func options(forRestorationId restorationId: String) -> AzureOptions? {
+        guard restorationId == "download" else { return nil }
+        return AppState.downloadOptions
+    }
+
+    func transfer(
+        _ transfer: Transfer,
+        didUpdateWithState _: TransferState,
+        andProgress progress: Float?
+    ) {
+        if let blobTransfer = transfer as? BlobTransfer {
+            print("BLOB DOWNLOAD \(blobTransfer.hash) DID UPDATE WITH PROGRESS %\((progress ?? -0.01) * 100)")
+            reloadTableView()
```

Comment:
Trying to selectively reload cells caused all kinds of visual and performance glitches. Then again, maybe I was just doing it wrong.

Reply:
Fixed.